### PR TITLE
Add API for adding unique particle

### PR DESCRIPTION
**Pull request by MuncleUscles**

### CODE REVIEW

File: src/modules/atommodel/particles/RadixUniqueParticle.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { RadixSerializer, RadixParticle, includeDSON, includeJSON, RadixAddress, RadixResourceIdentifier } from '..';
 
 @RadixSerializer.registerClass('UNIQUEIDPARTICLE')
-export class RadixTokenClassParticle extends RadixParticle {
+export class RadixUniqueParticle extends RadixParticle {
```
Comment (laszlourszuly) [inline src/modules/atommodel/particles/RadixUniqueParticle.ts]: 👍 This name is much, much better!

File: test/integration/UniqueParticle.spec.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                done()
+            },
+            error: e => done(new Error(JSON.stringify(e))),
+        })
+    })
+
+    it('(2) should fail submitting an atom with a conflicing unique id', function (done) {
```
Comment (laszlourszuly) [inline test/integration/UniqueParticle.spec.ts]: Nitpicking: Typo (`... conflicing ...` -> `... conflicting ...`)

File: test/integration/UniqueParticle.spec.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        .addUniqueParticle(identity1.account, 'unique3')
+        .signAndSubmit(identity1)
+        .subscribe({
+            complete: () => {
+                // Check balance
+                done()
+            },
+            error: e => done(new Error(JSON.stringify(e))),
+        })
+    })
```
Comment (laszlourszuly) [inline test/integration/UniqueParticle.spec.ts]: What would happen if we ran this test directly after the above one?

```javascript
    it('(3b) should succeed/fail submitting an atom with partially conflicting unique id', function (done) {
        this.timeout(50000)

         RadixTransactionBuilder.createMintAtom(
            identity1.account,
            testTokenRef,
            1)
        .addUniqueParticle(identity1.account, 'unique2')
        .signAndSubmit(identity1)
        .subscribe({
            complete: () => {
                // Check balance
                done()
            },
            error: e => done(new Error(JSON.stringify(e))),
        })
    })
```

I guess my real question is whether it's the combination of `unique`'s that is unique or each individual `unique`.

File: src/modules/atommodel/particles/RadixUniqueParticle.ts (hunk 1 above)
Comment (MuncleUscles) [reply src/modules/atommodel/particles/RadixUniqueParticle.ts]: I know, right? I was hit by a wave of inspiration and there it was

File: test/integration/UniqueParticle.spec.ts (hunk 3 above)
Comment (MuncleUscles) [reply test/integration/UniqueParticle.spec.ts]: It's each individual unique, so your test should fail. I'll add it as a real test

**Pull request merged**